paimon-core/src/main/java/org/apache/paimon/catalog/FileSystemCatalog.java

```diff
@@ -99,6 +99,24 @@ public void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade
         uncheck(() -> fileIO.delete(databasePath(name), true));
     }
 
+    @Override
+    public void renameDatabase(String fromDatabase, String toDatabase, boolean ignoreIfNotExists)
+            throws DatabaseNotExistException, DatabaseAlreadyExistException {
+        if (!databaseExists(fromDatabase)) {
+            if (ignoreIfNotExists) {
+                return;
+            }
+
+            throw new DatabaseNotExistException(fromDatabase);
+        }
+
+        if (databaseExists(toDatabase)) {
+            throw new DatabaseAlreadyExistException(toDatabase);
+        }
+
+        uncheck(() -> fileIO.rename(databasePath(fromDatabase), databasePath(toDatabase)));
```

Comment:
If there are several task that write the table data of this database, what's the behavior of these tasks?

Reply:
I think that if there are multiple tasks writing to this database, after modifying the database name, all tasks need to re-designate the database. Is there any way to make the writing task aware of the database name change?

Reply:
IMO, there is no way to aware of database name change, therefore the `renameDatabase` depend on the write path listener which we could introduce the listener to update the path of writer.

Reply:
ok, let me think about it.

Reply:
@s7monk, above problem could be solved in another pull request. This pull request could keep the same behavior of `renameTable`.